mail_tracking/models/res_partner.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antonio Espinosa - <antonio.espinosa@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models, api, fields
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    tracking_email_ids = fields.Many2many(
+        string="Tracking emails", comodel_name="mail.tracking.email",
+        readonly=True)
+    tracking_emails_count = fields.Integer(
+        string="Tracking emails count", store=True, readonly=True,
+        compute="_compute_tracking_emails_count")
+    email_score = fields.Float(
+        string="Email score",
+        compute="_compute_email_score", store=True, readonly=True)
+    email_stars = fields.Selection(
+        string="Email reputation", selection=[
+            ('0', 'None'),
+            ('1', 'Very Low'),
+            ('2', 'Low'),
+            ('3', 'Medium'),
+            ('4', 'High'),
+            ('5', 'Very high'),
+        ], compute="_compute_email_stars", store=True, readonly=True)
+
+    @api.one
+    @api.depends('tracking_email_ids.state')
+    def _compute_email_score(self):
+        self.email_score = self.env['mail.tracking.email'].email_score(
+            self.tracking_email_ids)
+
+    @api.one
+    @api.depends('email_score')
+    def _compute_email_stars(self):
+        self.email_stars = self.env['mail.tracking.email'].email_stars(
+            self.email_score)
+
+    @api.one
+    @api.depends('tracking_email_ids')
+    def _compute_tracking_emails_count(self):
+        self.tracking_emails_count = len(self.tracking_email_ids)
+
+    @api.multi
+    def write(self, vals):
+        email = vals.get('email')
+        if email:
```

Comment:
It should be `if email is not None:` and check if you are resetting the e-mail to False.